packages/zcli-apps/tests/functional/create.test.ts

```diff
@@ -34,26 +40,46 @@ describe('apps', function () {
           createAppPkgStub.onFirstCall().resolves('thePathLessFrequentlyTravelled')
           uploadAppPkgStub.onFirstCall().resolves({ id: 817 })
           uploadAppPkgStub.onSecondCall().resolves({ id: 818 })
-        })
-        .nock('https://z3ntest.zendesk.com/', api => {
-          api
-            .post('/api/apps.json', { upload_id: 817, name: 'Test App 1' })
-            .reply(200, { job_id: 127 })
-          api
-            .post('/api/apps.json', { upload_id: 818, name: 'Test App 2' })
-            .reply(200, { job_id: 128 })
-          api
-            .get('/api/v2/apps/job_statuses/127')
-            .reply(200, { status: 'completed', message: 'awesome', app_id: 123456 })
-          api
-            .get('/api/v2/apps/job_statuses/128')
-            .reply(200, { status: 'completed', message: 'awesome', app_id: 123458 })
-          api
-            .post('/api/support/apps/installations.json', { app_id: '123456', settings: { name: 'Test App 1' } })
-            .reply(200)
-          api
-            .post('/api/support/apps/installations.json', { app_id: '123458', settings: { name: 'Test App 2', salesForceId: 123 } })
-            .reply(200)
+
+          fetchStub.withArgs(sinon.match({
+            url: 'https://z3ntest.zendesk.com/api/apps.json'
+          })).onFirstCall().resolves({
+            body: JSON.stringify({ job_id: 127 }),
+            text: () => Promise.resolve(JSON.stringify({ job_id: 127 })),
+            ok: true
+          }).onSecondCall().resolves({
+            body: JSON.stringify({ job_id: 128 }),
+            text: () => Promise.resolve(JSON.stringify({ job_id: 128 })),
+            ok: true
+          })
+          fetchStub.withArgs(sinon.match({
+            url: 'https://z3ntest.zendesk.com/api/v2/apps/job_statuses/127'
+          })).resolves({
+            text: () => Promise.resolve(JSON.stringify({ status: 'completed', message: 'awesome', app_id: 123456 })),
+            body: JSON.stringify({ status: 'completed', message: 'awesome', app_id: 123456 }),
+            ok: true
+          })
+          fetchStub.withArgs(sinon.match({
+            url: 'https://z3ntest.zendesk.com/api/v2/apps/job_statuses/128'
+          })).resolves({
+            text: () => Promise.resolve(JSON.stringify({ status: 'completed', message: 'awesome', app_id: 123458 })),
+            body: JSON.stringify({ status: 'completed', message: 'awesome', app_id: 123458 }),
+            ok: true
+          })
+          fetchStub.withArgs(sinon.match({
+            url: 'https://z3ntest.zendesk.com/api/support/apps/installations.json'
```

Comment:
Do we need a `method: 'POST'` for these requests?

Reply:
Never mind about this question, just learned that we don't need to explicitly specify the HTTP method (e.g., POST, GET, etc.) because the stubbed function itself (in this case, fetch) is being replaced by a Sinon stub.